internal/loraserver/loraserver.go

```diff
@@ -88,6 +98,24 @@ func handleRXPackets(ctx Context) {
 	wg.Wait()
 }
 
+func handleTXMACPayloads(ctx Context) {
+	var wg sync.WaitGroup
+	for txMACPayload := range ctx.Controller.TXMACPayloadChan() {
+		go func(txMACPayload models.MACPayload) {
+			wg.Add(1)
+			if err := addTXMACToQueue(ctx, txMACPayload); err != nil {
+				log.WithFields(log.Fields{
+					"dev_eui":     txMACPayload.DevEUI,
+					"reference":   txMACPayload.Reference,
+					"data_base64": base64.StdEncoding.EncodeToString(txMACPayload.MACCommand),
+				}).Errorf("add tx mac-payload to queue error: %s", err)
+			}
+			wg.Done()
```

Comment:
nit: isn't it better to move this right after `wg.Add(1)` with a `defer`? Otherwise a tiny refactoring might break the logic.